balancer/backends.go

```diff
@@ -110,3 +123,35 @@ func (b *backends) connect(addr string) error {
 	b.mu.Unlock()
 	return nil
 }
+
+func (b *backends) loop(queryInterval time.Duration) {
+	t := time.NewTicker(queryInterval)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-b.closing:
+			close(b.closed)
+			return
+		case <-t.C:
+			if err := b.updateBackendScores(); err != nil {
+				grpclog.Printf("failed to update backend load scores: %s", err)
+			}
+		}
+	}
+}
+
+func (b *backends) updateBackendScores() error {
+	b.mu.RLock()
+	set := b.set
+	b.mu.RUnlock()
+
+	succeeded := make([]string, 0, len(set))
+	for addr, backend := range set {
+		if err := backend.UpdateScore(); err != nil {
```

Comment:
I think you need to have some grace-period, or a number of failures, etc before evicting a backend. at least, you should check the error with `grpc.Code(err)` and make a decision based on that. the main concern is that e.g. on a temporary network problem, you will simply evict all backends